syft/pkg/cataloger/binary/cataloger.go

```diff
@@ -12,16 +12,42 @@ import (
 
 const catalogerName = "binary-cataloger"
 
-func NewCataloger() pkg.Cataloger {
-	return &Cataloger{}
+type CatalogerConfig struct {
+	classifiers []classifier
+	overwrite   bool
```

Comment:
from the community meeting: we can probably remove the override option here, export the default classifiers in a separate function (from https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/binary/default_classifiers.go ), and export the classifier struct https://github.com/anchore/syft/blob/4c20a74d2fc6774b7745b432995d5ff7836b90a0/syft/pkg/cataloger/binary/classifier.go#L27 and evidenceMatcher type.